api/v1beta1/cluster_types.go

```diff
@@ -72,6 +72,24 @@ type ClusterSpec struct {
 	// this feature is highly experimental, and parts of it might still be not implemented.
 	// +optional
 	Topology *Topology `json:"topology,omitempty"`
+
+	// availabilityGates specifies additional conditions to include when evaluating Cluster availability.
+	// A Cluster is available if:
+	// * Cluster's `RemoteConnectionProbe` and `TopologyReconciled` conditions are true and
+	// * the control plane `Available` condition is true and
+	// * all worker resource's `Available` conditions are true and
+	// * all conditions defined in AvailabilityGates are true as well.
+	// +optional
+	// +listType=map
+	// +listMapKey=conditionType
+	AvailabilityGates []ClusterAvailabilityGate `json:"availabilityGates,omitempty"`
+}
+
+// ClusterAvailabilityGate contains the type of a Cluster condition to be used as availability gate.
+type ClusterAvailabilityGate struct {
+	// conditionType refers to a condition with matching type in the Cluster's condition list.
+	// Note: Both Cluster API conditions or conditions added by 3rd party controllers can be used as availability gates.
+	ConditionType string `json:"conditionType"`
```

Comment:
This is a required field? Please mark it explicitly as required

Also, can we add a maximal length to this field? And a minimum? We should check to see if there are any validations on the metav1.Condition.Type field.

If we can't justify a particular maximum length, pick something large enough we should never hit it (I can't imagine a ConditionType would be over 256 chars would it? Given the list of known conditions we could even go down lower than that)

For minimum length, `""` would be valid without a minimum length and we probably don't want that, so, at least 1 for minimum length forces some choice for the value.

Reply:
fixed (used validation rules from condition types + minlength)